src/frontend/test_runner/tests/testdata/tpch.yaml

```diff
@@ -242,90 +258,83 @@
                         BatchExchange { order: [], dist: HashShard(partsupp.ps_suppkey) }
                           BatchHashJoin { type: Inner, predicate: partsupp.ps_partkey = part.p_partkey, output: [partsupp.ps_suppkey, partsupp.ps_supplycost, part.p_partkey, part.p_mfgr] }
                             BatchExchange { order: [], dist: HashShard(partsupp.ps_partkey) }
-                              BatchScan { table: partsupp, columns: [partsupp.ps_partkey, partsupp.ps_suppkey, partsupp.ps_supplycost], distribution: SomeShard }
-                            BatchExchange { order: [], dist: HashShard(part.p_partkey) }
-                              BatchProject { exprs: [part.p_partkey, part.p_mfgr] }
-                                BatchFilter { predicate: (part.p_size = 4:Int32) AND Like(part.p_type, '%TIN':Varchar) }
-                                  BatchScan { table: part, columns: [part.p_partkey, part.p_mfgr, part.p_type, part.p_size], distribution: SomeShard }
+                              BatchScan { table: partsupp, columns: [partsupp.ps_partkey, partsupp.ps_suppkey, partsupp.ps_supplycost], distribution: UpstreamHashShard(partsupp.ps_partkey, partsupp.ps_suppkey) }
+                            BatchProject { exprs: [part.p_partkey, part.p_mfgr] }
+                              BatchFilter { predicate: (part.p_size = 4:Int32) AND Like(part.p_type, '%TIN':Varchar) }
+                                BatchScan { table: part, columns: [part.p_partkey, part.p_mfgr, part.p_type, part.p_size], distribution: UpstreamHashShard(part.p_partkey) }
```

Comment:
Looks like we should also insert an exchange here? cc @st1page @xxchan

Reply:
strange... I will fix it.

Reply:
https://github.com/singularity-data/risingwave/pull/4904